pkg/network/node/networkpolicy.go

```diff
@@ -1190,3 +1186,10 @@ func (np *networkPolicyPlugin) refreshPodNetworkPolicies(pod *corev1.Pod) bool {
 func getPodFullName(pod *corev1.Pod) string {
 	return fmt.Sprintf("%s/%s", pod.Namespace, pod.Name)
 }
+
+func (npns *npNamespace) GetMatchVNID(vnid uint32) uint32 {
+	if npns.name == HostNetworkNamespace {
+		return 0
+	}
+	return vnid
```

Comment:
you can just return `npns.vnid` rather than taking a `vnid` argument. (And then just update the loop in `selectNamespacesInternal` to be `for _, npns := range np.namespaces`

Reply:
I was thinking about that, but then I decided not to touch anything else. Sure, I will add that change.